evalml/pipelines/components/transformers/imputers/target_imputer.py

```diff
@@ -124,19 +129,11 @@ def transform(self, X, y):
         transformed = self._component_obj.transform(y_df)
         y_t = pd.Series(transformed[:, 0], index=y_ww.index)
 
-        # TODO: Fix this after WW adds inference of object type booleans to BooleanNullable
-        # Iterate through categorical columns that might have been boolean and convert them back to boolean
-        if {True, False}.issubset(set(y_t.unique())) and isinstance(
-            y_ww.ww.logical_type,
-            Categorical,
-        ):
-            y_t = y_t.astype(bool)
-
-        new_logical_type = (
-            Integer
-            if isinstance(y_ww.ww.logical_type, IntegerNullable)
-            else y_ww.ww.logical_type
-        )
+        new_logical_type = y_ww.ww.logical_type
+        if isinstance(y_ww.ww.logical_type, IntegerNullable):
+            new_logical_type = Double
```

Comment:
Converting imputed IntegerNullable data to Integer truncates the imputed values for mean and median, so I changed this to Double, which matches other imputers' behavior.